### PR TITLE
[MIRROR] Adds the DNA Infuser, a genetics machine you feed corpses to infuse their DNA with yours! What could go wrong?! [MDB IGNORE]

### PR DESCRIPTION
**Pull request by SkyratBot**
Original PR: https://github.com/tgstation/tgstation/pull/71351
--------------------
## About The Pull Request  
Adds the "DNA Infuser" to genetics. One person enters, a corpse is added to the machine, and you can activate the machine to "infuse" the subject with the DNA. This converts one random organ from a set into the mob-related organ.

### Rat mutation 🐀

Rats can be fed in to turn you into a rat-creature-thing!
```diff
+See better in the dark
+Can pretty much eat anything! Toxic foods, gross foods, whatever works!
+Smaller, and can climb tables
?Randomly squeaks occasionally?
-Take twice as much damage
-Vulnerable to flashes
-Gets hungry MUCH quicker.
-Yes, eat anything, but only ENJOY dairy.
```
Having every rat organ at once allows you to ventcrawl nude!

### Carp mutation 🐟 

Carp work for a mutation as well!
```diff
+Strong jaws, that drop teeth over time!
+Space immunity! Breathe in space, unbothered by pressure or cold!
+Smaller, and can climb tables
-Can't block your jaws with a mask
-Can't take the heat, overheats easily
-Can only breathe in environments that have minimal or no oxygen
-Nomadic. If you don't enter a new zlevel for awhile, you'll start feeling anxious.
```
Having every carp organ at once allows you to swim through space!

### Fly mutation 🪰 

Any corpses without organs to turn into turn into fly organs! Fly organs now have a bonus for collecting them all, transforming you into a fly, when you pass the threshold. But even without those, fly organs are technically... organs. They most of the time work like normal ones.

## Todo 🐦 

- [x] Finish the infuser code
- [x] Create a little booklet that shows what kind of shit you can turn into, hopefully i can autogenerate this based off of organ set subtypes list
- [x] sprite/slap a color on rat mutant organs
- [x] Maybe make a *few* more organ sets

## Why It's Good For The Game 🐑 

Oops, I forgor to fill this out! My hackmd is here.

https://hackmd.io/@ bazelart/ByFkhuUIi

## Changelog 🧬 

:cl: Tralezab code, Azlan + Azarak (Az gaaang) for the organs
add: Added the DNA infuser to genetics! Person goes in, corpse goes in, and they combine!
add: Try not to turn yourself into a fly, OK?
/:cl:

### CODE REVIEW

**Comment by Tastyfish** [comment]
Fixed actual conflicts but I'm frightened to imagine how this will work with our bodypart code.

File: code/datums/mood_events/generic_negative_events.dm [hunk 1]
```diff
@@ -421,8 +421,17 @@
 	mood_change = -2
 	timeout = 4 MINUTES
 
+/datum/mood_event/sabrage_fail
```
Comment (Funce) [inline code/datums/mood_events/generic_negative_events.dm]: Duplicate Sabrage Event

**Review by Funce — CHANGES_REQUESTED** [review]

**Review by Funce — COMMENTED** [review]
No_equip flags are fine, all special organs have been moved, if someone tests it locally on Skyrat Code I'd feel safe merging it

**Review by Funce — APPROVED** [review]

**Review by tf-4 — APPROVED** [review]
seems to work fine to me

**Pull request merged**